docs/general/developer-resources/emerald-paratime/writing-dapps-on-emerald.md

```diff
@@ -614,6 +614,107 @@ share them with us at [#dev-support on our Community Slack channel][slack-devsup
 [metamask]: ../../manage-tokens/how-to-transfer-rose-into-emerald-paratime.mdx#verifying-rose-balance-on-emerald-paratime
 [slack-devsupport]: https://oasiscommunity.slack.com/messages/dev-support
 
+## Create Vyper dApp on Emerald with Rust
+
+Though Solidity is usually the most popular choice when it comes to smart contracts,
+[vyper](https://vyper.readthedocs.io/en/stable/) is also a modern and popular alternative.
+As Emerald provides full EVM support, Vyper is a viable choice for your Emerald dApps.
+Apart from the choice of programming language when it comes to smart contracts, you also
+are free to pick alternatives of JavaScript/TypeScript when it comes to deploying or
+calling your dApps. For example, you can also use the
+[Rust version of ethers](https://github.com/gakonst/ethers-rs/) to deploy your dApps to
+Emerald.
+
+Let's consider the following example of a small Vyper smart contract:
+```python
+greeting: String[100]
+
+@external
+def __init__(_greeting: String[80]):
+  self.greeting = concat("HELLO FROM VYPER! ", _greeting)
+
+@external
+def greet() -> String[100]:
+  return self.greeting
+
+@external
+def setGreeting(_greeting: String[100]):
+  self.greeting = _greeting
+```
+
+If this is your first experience with Vyper, you must first install the compiler. The
+best practice is to install it via [pipx](https://pypi.org/project/pipx/):
+`pipx install vyper`, so that it runs from an isolated environment. After this, you
+can compile your ABI and your bytecode by running the following bash commands:
+
+```bash
+# considering you save the smart contract to greeter.vy, this will compile the abi:
+~/.local/pipx/venvs/vyper/bin/vyper -f abi greeter.vy > greeter_abi.json
+# and this command will allow you to compile the bytecode:
+~/.local/pipx/venvs/vyper/bin/vyper -f bytecode greeter.vy > greeter_bytecode
+```
+
+The ABI contains an instruction set that you can import in any client that will call
+your smart contract, and the bytecode contains the actual contract that will get
+executed by the blockchain.
+
+Now you that have these two files, you can proceed to deploy.
+If you would like to stick to Hardhat, you first have to install
+[the plugin](https://hardhat.org/plugins/nomiclabs-hardhat-vyper.html) and after that
+you can proceed with the normal usage of Hardhat. In case you are looking for alternatives,
+please find below an example of how to deploy the vyper smart contract by using Rust:
```

Comment:
Write how to actually deploy the smart contract with Rust? For example, what goes into Cargo.toml and some initial commands to set up the Rust project.